### PR TITLE
ddl: fix modify column null to not null bug

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
When modify column from null to not null, the normal state should be:
```
column null -> check null value and set column prevent not null -> column not null
```

But there is something wrong that `set column prevent not null` does not take effect, so the problem is:
```
column null -> check null value and change column not null
```
there is only 2 state. here is the problem.

This PR will fix this.

### Check List 

Tests 

 - Unit test
Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -380,15 +380,10 @@ func (w *worker) doModifyColumn(t *meta.Meta, job *model.Job, newCol *model.Colu
...
-		ver, err = modifyColumnFromNull2NotNull(w, t, dbInfo, tblInfo, job, oldCol, newCol)
-		if err != nil {
-			return ver, errors.Trace(err)
-		}
+	if !mysql.HasNotNullFlag(oldCol.Flag) && mysql.HasNotNullFlag(newCol.Flag) && !mysql.HasPreventNullInsertFlag(oldCol.Flag) {
 		// Introduce the `mysql.HasPreventNullInsertFlag` flag to prevent users from inserting or updating null values.
-		if !mysql.HasPreventNullInsertFlag(oldCol.Flag) {
-			return ver, nil
-		}
+		ver, err = modifyColumnFromNull2NotNull(w, t, dbInfo, tblInfo, job, oldCol, newCol)
```
Comment (winkyao) [inline ddl/column.go]: When will remove the mysql.PreventNullInsertFlag?

Comment (crazycs520) [reply ddl/column.go]: see line 425, 436.

**Comment by winkyao** [comment]
Please cherry pick this PR to 2.1 branch.

**Pull request merged**